plugin-server/src/utils/db/db.ts

```diff
@@ -194,12 +200,23 @@ export class DB {
 
     // Postgres
 
-    public postgresQuery<R extends QueryResultRow = any, I extends any[] = any[]>(
+    public async postgresQuery<R extends QueryResultRow = any, I extends any[] = any[]>(
```

Comment:
I do not think we should put the caching logic into `postgresQuery` itself

Reply:
Where else would we put this? I think the benefit here is that we can easily cache any query we want to in the future, rather than handling individually with each query.

As for simply extracting this logic elsewhere and having this function call it would end up looking the same - or what are you suggesting?

Reply:
I think each function should do the min amount of work possible. This usually limits scope and blast radius if something doesn't go as expected

Reply:
we can have a separate function `postgresQueryWithCaching` or whatever that has the caching part and calls this function as it was if needed.

Reply:
I'll look at a refactor. These will inevitably be tightly coupled though, but agree with both that we can probably do better here